src/test/java/com/example/onlinebookstore/service/BookServiceTest.java

```diff
@@ -0,0 +1,202 @@
+
+@ExtendWith(MockitoExtension.class)
+public class BookServiceTest {
+    private static Book book;
+    private static CreateBookRequestDto createBookRequest;
+    private static BookSearchParametersDto searchParameters;
+    private static BookDto bookDto;
+    @InjectMocks
+    private BookServiceImpl bookService;
+    @Mock
+    private BookRepository bookRepository;
+    @Mock
+    private BookMapper bookMapper;
+    @Mock
+    private SpecificationBuilder<Book> specificationBuilder;
+
+    @BeforeAll
+    static void beforeAll() {
+        bookDto = new BookDto();
+        bookDto.setId(1L);
+        bookDto.setTitle("Example Book");
+        bookDto.setAuthor("John Doe");
+        bookDto.setIsbn("978-1234567890");
+        bookDto.setPrice(new BigDecimal("19.99"));
+        bookDto.setDescription("An example book description.");
+        bookDto.setCoverImage("book_cover.jpg");
+        Set<Long> categoryIds = new HashSet<>();
+        categoryIds.add(101L);
+        categoryIds.add(102L);
+        bookDto.setCategoriesIds(categoryIds);
+
+        book = new Book();
+        book.setTitle("The Great Gatsby");
+        book.setAuthor("F. Scott Fitzgerald");
+        book.setIsbn("978-3-16-148410-0");
+        book.setPrice(new BigDecimal("19.99"));
+        book.setDescription("A classic novel about the Jazz Age");
+        book.setCoverImage("great_gatsby.jpg");
+        Set<Category> categories = new HashSet<>();
+        categories.add(new Category());
+        book.setCategories(categories);
+
+        createBookRequest = new CreateBookRequestDto();
+        createBookRequest.setTitle("Pride and Prejudice");
+        createBookRequest.setAuthor("Jane Austen");
+        createBookRequest.setIsbn("978-0-14-143951-8");
+        createBookRequest.setPrice(new BigDecimal("12.99"));
+        createBookRequest.setDescription("A classic romance novel");
+        createBookRequest.setCoverImage("pride_prejudice.jpg");
+        Set<Long> categoryIdsForDto = new HashSet<>();
+        categoryIdsForDto.add(1L);
+        categoryIdsForDto.add(2L);
+        createBookRequest.setCategoryIds(categoryIdsForDto);
+
+        String[] authors = {"Author 1", "Author 2"};
+        String[] titles = {"Title 1", "Title 2"};
+        searchParameters = new BookSearchParametersDto(authors, titles);
+    }
+
+    @Test
+    @DisplayName("getBookById_returnsBook_ok")
+    public void getBookById_returnsBook_ok() {
+        when(bookRepository.findByIdBookAndItsCategories(anyLong())).thenReturn(Optional.of(book));
+        when(bookMapper.toDto(any(Book.class))).thenReturn(bookDto);
+
+        assertEquals(bookDto, bookService.getBookById(1L));
+        verify(bookMapper, times(1)).toDto(any(Book.class));
+        verify(bookRepository, times(1)).findByIdBookAndItsCategories(anyLong());
+    }
+
+    @Test
+    @DisplayName("getBookById_NonExistingId_throwsException_notOk")
+    public void getBookById_NonExistingId_throwsException_notOk() {
```

Comment:
you do have not consistent method naming for test cases. The method contains 4 parts, but others have 3 parts. You can use the naming convention that Bohdan showed us in the video. For example, u can name this method like `getBookById_withValidId_throwsException`. Please, check it for all other methods